tests/test_wasting.py

```diff
@@ -0,0 +1,838 @@
+                 healthseekingbehaviour.HealthSeekingBehaviour(resourcefilepath=resourcefilepath),
+                 healthburden.HealthBurden(resourcefilepath=resourcefilepath),
+                 simplified_births.SimplifiedBirths(resourcefilepath=resourcefilepath),
+                 wasting.Wasting(resourcefilepath=resourcefilepath)
+                 )
+    return sim
+
+
+@pytest.mark.slow
+def test_basic_run(tmpdir):
+    """Run the simulation and do some daily checks on dtypes and properties integrity """
+    class DummyModule(Module):
+        """ A Dummy module that ensure wasting properties are as expected on a daily basis """
+        METADATA = {Metadata.DISEASE_MODULE}
+
+        def read_parameters(self, data_folder):
+            pass
+
+        def initialise_population(self, population):
+            pass
+
+        def initialise_simulation(self, sim):
+            # schedule check property integrity event
+            sim.schedule_event(CheckPropertyIntegrityEvent(self), sim.date)
+
+        def on_birth(self, mother_id, child_id):
+            pass
+
+    class CheckPropertyIntegrityEvent(RegularEvent, PopulationScopeEventMixin):
+        def __init__(self, module):
+            """schedule to run every day
+            :param module: the module that created this event
+            """
+            self.repeat_days = 1
+            super().__init__(module, frequency=DateOffset(days=self.repeat_days))
+            assert isinstance(module, DummyModule)
+
+        def apply(self, population):
+            """ Apply this event to the population.
+               :param population: the current population
+           """
+            # check datatypes
+            self.check_dtypes(population)
+
+            # check properties are as expected
+            self.check_configuration_of_properties(population)
+
+        def check_dtypes(self, population):
+            # Check types of columns
+            df = population.props
+            orig = population.new_row
+            assert (df.dtypes == orig.dtypes).all()
+
+        def check_configuration_of_properties(self, population):
+            """ check wasting properties on a daily basis to ensure integrity """
+            df = population.props
+            under5_sam = df.index[df.is_alive & (df.age_exact_years < 5) &
+                                  (df.un_clinical_acute_malnutrition == 'SAM')]
+
+            # Those that were never wasted, should have normal WHZ score:
+            assert (df.loc[~df.un_ever_wasted & ~df.date_of_birth.isna(), 'un_WHZ_category'] == 'WHZ>=-2').all()
+
+            # Those for whom the death date has past should be dead
+            assert not df.loc[df.un_ever_wasted & (df['un_sam_death_date'] < self.sim.date), 'is_alive'].any()
+            assert not df.loc[
+                (df.un_clinical_acute_malnutrition == 'SAM') & (df['un_sam_death_date'] < self.sim.date),
+                'is_alive'].any()
```

Comment:
@mnjowe, @thewati, Is there any reason why there is not:
`assert not df.loc[(df['un_sam_death_date'] < self.sim.date), 'is_alive'].any()`
instead of the two lines above?

Reply:
Not that I am aware of...  but it makes sense that it should just be `assert` without the `not`.

Reply:
@thewati the `not` is important as we want to ensure no individual who has their date of death from SAM recorded in simulation are alive

@EvaJanouskova I think both are important unless the code has changed as I haven't taken a look for some time. The first assert in addition to checking no individual who has their date of death recorded in the simulation is alive is also trying to ensure those who have their date of death recorded should also be True to ever having a history of wasting. Same with the second assert which is ensuring that anyone who has the date of death recorded in SAM column should also have SAM as a diagnosis

Reply:
It actually doesn't have to be true, that is why. Because the `un_ever_wasted` is indicator (at least with how it is coded) of whether they ever had low WHZ, but they still can have MAM or SAM due to low MUAC and/or because they have nutritional oedema, while the WHZ is normal.

Reply:
So I think, it should be as follows then:
            # Those for whom the death date has past should be dead
            assert not df.loc[(df['un_sam_death_date'] < self.sim.date), 'is_alive'].any()
            # Those who died due to SAM should have SAM
            assert (df.loc[(df['un_sam_death_date']) < self.sim.date, 'un_clinical_acute_malnutrition'] == 'SAM').all()

@mnjowe, @thewati, Do you agree?

Reply:
@timcolbourn, What was the idea behind `un_ever_wasted`, would you know? It is used in _stunting_ module as predictor in the linear model for probability per year of becoming stunted, and in the linear model for the probability per year of progressing from a state of being non-severely stunted to severe stunting. Should it be based on ever having lower WHZ, or ever having acute malnutrition (MAM or SAM)?

Reply:
As this property `un_ever_wasted` is indeed only concerned with WHZ not diagnosis, I agree with you @EvaJanouskova. You can proceed with the tests as suggested